DRILL-6711: Use jitpack repository for Drill Calcite project artifacts instead of repository.mapr.com (#1815)
apache/drill: docs/dev/Calcite.md

# Drill-specific commits in Apache Calcite

Currently, Drill uses Apache Calcite with additional changes, required for Drill. All the commits were left after 

> update from Calcite `1.4.0` to Calcite `1.15.0` and weren't merged to the Calcite's master yet.

Reviewer: ```suggestion
update from Calcite `1.4.0` to Calcite `1.15.0` and weren't merged to the Calcite's master yet since there is no consensus on them in Calcite community.
```

List of Jiras with Drill-specific commits:

|Jira|Summary|The reason why it wasn't merged|
|----|-------|-------------------------------|

> |[CALCITE-2018](https://issues.apache.org/jira/browse/CALCITE-2018)|Queries failed with AssertionError: rel has lower cost than best cost of subset|Pull request with the fix was created ([PR-552](https://github.com/apache/calcite/pull/552)), but an issue which blocks it was found and wasn't resolved yet: [CALCITE-2166](https://issues.apache.org/jira/browse/CALCITE-2166).|

Reviewer: ```suggestion
|[CALCITE-2018](https://issues.apache.org/jira/browse/CALCITE-2018)|Queries failed with AssertionError: rel has lower cost than best cost of subset|Pull request with the fix was created ([PR-552](https://github.com/apache/calcite/pull/552)), but [CALCITE-2166](https://issues.apache.org/jira/browse/CALCITE-2166) which blocks it was found and is not resolved yet.|
```

|[CALCITE-2087](https://issues.apache.org/jira/browse/CALCITE-2087)|Add new method to ViewExpander interface to allow passing SchemaPlus.|Pull request into Apache Calcite was created, but it was declined. See conversation in Jira.|

> |[CALCITE-1178](https://issues.apache.org/jira/browse/CALCITE-1178)|Allow SqlBetweenOperator to compare DATE and TIMESTAMP|SQL spec does not allow to compare datetime types if they have different <primary datetime field>s. Therefore Calcite community won’t accept these changes. Similar issues were reported in [CALCITE-2829](https://issues.apache.org/jira/browse/CALCITE-2829) and in [CALCITE-2745](https://issues.apache.org/jira/browse/CALCITE-2745)|

Reviewer: ```suggestion
|[CALCITE-1178](https://issues.apache.org/jira/browse/CALCITE-1178)|Allow SqlBetweenOperator to compare DATE and TIMESTAMP|SQL spec does not allow to compare datetime types if they have different <primary datetime field>s. Therefore Calcite community won’t accept these changes. Similar issues were reported in [CALCITE-2829](https://issues.apache.org/jira/browse/CALCITE-2829) and in [CALCITE-2745](https://issues.apache.org/jira/browse/CALCITE-2745).|
```

# Drill-Calcite repository

Repository with source code is placed [here](https://github.com/vvysotskyi/drill-calcite). For backward 

> compatibility, some previously used by Drill branches were pushed to this repo.

Reviewer: ```suggestion
compatibility, a couple of previous Drill Calcite branches were pushed into this repo.
```

> For the case when committer should be provided with required rights, please notify repository owner about that.

Reviewer: ```suggestion
Drill committers who need write permissions to the repository, should notify its owner.
```

# Process of updating Calcite version

> - Push required changes to the branch, or create required a branch. *Though this repository contains Drill specific commits, it is forbidden to add additional specific commits which weren't merged to Apache Calcite master if it wasn't discussed broadly.*

Reviewer: ```suggestion
- Push required changes to the existing branch, or create new branch. *Though this repository contains Drill specific commits, it is forbidden to add additional specific commits which were not merged to Apache Calcite master if it wasn't discussed in Drill community first.*
```

> - The last commit should be a commit with updating the version number.

Reviewer: ```suggestion
- The last commit must be a commit which updates the version number.
```

> - Create and push tag for commit with a version update. Tag name should match the version, for example, `1.18.0-drill-r2`.

Reviewer: ```suggestion
- Create and push tag for commit with the version update. Tag name should match the version, for example, `1.18.0-drill-r0`, `1.18.0-drill-r1`, `1.18.0-drill-r2` and so on.
```

> - Bump-up Calcite version in Drill.

Reviewer: ```suggestion
- Bump-up Drill Calcite version in Drill pom.xml file.
```
